plant/plants/api/views.py

```diff
@@ -28,6 +44,47 @@ def get_queryset(self):
     def perform_create(self, serializer):
         serializer.save(owner=self.request.user)
 
+    @action(detail=True, methods=["GET"])
+    def plant_stats(self, request, pk=None):
+        start_date = request.query_params.get("start_date")
+        end_date = request.query_params.get("end_date")
+        plant = get_object_or_404(Plant, pk=pk)
+        the_most_active_users = (
+            plant.waterings.values("user")
+            .order_by("user")
+            .annotate(total_litres=Sum("litres"))[:3]
+        )
+        average_watering_per_month = plant.waterings.values(
+            month=F("watering_date__month")
+        ).annotate(average_litres=Avg("litres"))
+
+        print(start_date, end_date)
+
+        if start_date and end_date:
+            waterings_count = plant.waterings.filter(
+                watering_date__range=(start_date, end_date)
+            ).count()
+        else:
+            waterings_count = plant.waterings.filter(
+                watering_date__lte=datetime.now() - timedelta(days=30)
+            ).count()
+
+        average_serializer = AverageWateringPerMonthSerializer(
```

Comment:
Please run pre-commit, so ruff will format the code, as I am sure splitting this instruction to multiple lines is completely unnecessary :D